spaceward/src/def-hooks/walletContext.tsx

```diff
@@ -2,115 +2,309 @@ import { createContext, ReactNode, useContext, useState } from "react";
+			}
+			const [account] = await client.signer.getAccounts();
+			wallet.accounts.push({
+				address: account.address,
+				pathIncrement: null,
+			});
 
-    try {
-      const wallet: Wallet = {
-        name: "Keplr Integration",
-        mnemonic: null,
-        HDpath: null,
-        password: null,
-        prefix: client.env.prefix ?? "warden",
-        pathIncrement: null,
-        accounts: [],
-      };
-      await client.useKeplr({
-        chainName: env.chainName,
-        bip44: {
-          coinType: 118,
-        },
-        currencies: [
-          {
-            coinDenom: "WARD",
-            coinMinimalDenom: "uward",
-            coinDecimals: 6,
-          },
-        ],
-        feeCurrencies: [
-          {
-            coinDenom: "WARD",
-            coinMinimalDenom: "uward",
-            coinDecimals: 6,
-            gasPriceStep: {
-              low: 25,
-              average: 50,
-              high: 100,
-            },
-          },
-        ],
-        stakeCurrency: {
-          coinDenom: "WARD",
-          coinMinimalDenom: "uward",
-          coinDecimals: 6,
-        },
-      });
-      if (!client.signer) {
-        throw new Error("client.signer undefined");
-      }
-      const [account] = await client.signer.getAccounts();
-      wallet.accounts.push({ address: account.address, pathIncrement: null });
+			setActiveWallet(wallet);
+			window.localStorage.setItem("lastWallet", wallet.name);
+			// if (activeWallet && activeWallet.name && activeWallet.password) {
+			//   setWallets([
+			//     ...wallets,
+			//     {
+			//       name: activeWallet.name,
+			//       wallet: CryptoJS.AES.encrypt(JSON.stringify(activeWallet), activeWallet.password).toString(),
+			//     },
+			//   ]);
+			// }
+			if (
+				activeWallet?.name == "Keplr Integration" &&
+				!activeWallet.password
+			) {
+				setWallets([
+					...wallets,
+					{
+						name: activeWallet.name,
+						wallet: JSON.stringify(activeWallet),
+					},
+				]);
+			}
 
-      setActiveWallet(wallet);
-      window.localStorage.setItem("lastWallet", wallet.name);
-      // if (activeWallet && activeWallet.name && activeWallet.password) {
-      //   setWallets([
-      //     ...wallets,
-      //     {
-      //       name: activeWallet.name,
-      //       wallet: CryptoJS.AES.encrypt(JSON.stringify(activeWallet), activeWallet.password).toString(),
-      //     },
-      //   ]);
-      // }
-      if (activeWallet?.name == "Keplr Integration" && !activeWallet.password) {
-        setWallets([
-          ...wallets,
-          {
-            name: activeWallet.name,
-            wallet: JSON.stringify(activeWallet),
-          },
-        ]);
-      }
+			setActiveClient(client);
+		} catch (e) {
+			console.error(e);
+		}
+		window.localStorage.setItem("wallets", JSON.stringify(wallets));
+	};
+	const connectWithLeap = async () => {
+		const client = useClient();
+		const chainId = "wardenprotocol";
```

Comment:
let's move this string into `env.ts` or this won't work in testnet, follow the same pattern of the other vars you'll find in that file